### PR TITLE
Added powered by giphy attribution mark

### PR DESCRIPTION
**Pull request by fred-bowker**

### CODE REVIEW

**Comment by fred-bowker** [comment]
When moving from beta to production key with the GIPHY API, you are required to display an attribution mark "Powered By Giphy"

https://support.giphy.com/hc/en-us/articles/360035158592-What-conditions-does-my-app-project-need-to-meet-in-order-to-get-a-production-API-Key-

It is not that clear where you need to display this:
https://github.com/buddyboss/buddyboss-platform/issues/1283

This pull request adds the attribution mark to the giphy popup, and is fairly inconspicuous, it is off by default as the app developer may already be displaying the attribution mark somewhere else in the app.

Basically I have had to add this to get through the review process to move from beta to production key

**Comment by fred-bowker** [comment]
@jaesung-0o0 Hi could you have a look at this pull request, many thanks Fred

File: Package.swift [hunk 1]
```diff
@@ -18,15 +18,19 @@ let package = Package(
...
+              ],
+              resources: [
+                   .process("Resources")
+                 ]
+          ),
+          .testTarget(
+              name: "ChatUITests",
+              dependencies: ["ChatUI"]),
+      ]
 )
```
Comment (jaesung-0o0) [inline Package.swift:36]:
```suggestion
        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
        // Targets can depend on other targets in this package, and on products in packages this package depends on.
        .target(
            name: "ChatUI",
            dependencies: [
                .product(name: "GiphyUISDK", package: "giphy-ios-sdk")
            ],
            resources: [
                .process("Resources")
            ]
        ),
        .testTarget(
            name: "ChatUITests",
            dependencies: ["ChatUI"]),
    ]
```

File: Package.swift [hunk 2]
```diff
@@ -18,15 +18,19 @@ let package = Package(
...
+                  .product(name: "GiphyUISDK", package: "giphy-ios-sdk")
+              ],
+              resources: [
+                   .process("Resources")
+                 ]
+          ),
+          .testTarget(
+              name: "ChatUITests",
+              dependencies: ["ChatUI"]),
+      ]
```
Comment (x-0o0) [inline Package.swift:35]:
```suggestion
        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
        // Targets can depend on other targets in this package, and on products in packages this package depends on.
        .target(
            name: "ChatUI",
            dependencies: [
                .product(name: "GiphyUISDK", package: "giphy-ios-sdk")
            ],
            resources: [
                .process("Resources")
            ]
        ),
        .testTarget(
            name: "ChatUITests",
            dependencies: ["ChatUI"]),
    ]
```

Comment (x-0o0) [reply Package.swift:35]: **Convention**: Indent -> 4 space

File: Sources/ChatUI/ChatInChannel/MessageField/Giphy/View/GiphyAttributionMarkView.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+class CustomTabBarController: UITabBarController {
+    var tabBarHeight: CGFloat {
+        tabBar.frame.size.height
+    }
+}
+
+struct GiphyAttributionMarkView: View {
+    
+    private let PADDING_ABOVE_TAB = 2.5
```
Comment (x-0o0) [inline Sources/ChatUI/ChatInChannel/MessageField/Giphy/View/GiphyAttributionMarkView.swift:11]:
```suggestion
    private let paddingAboveTab = 2.5
```
please modify snake case to lower camel case :)

File: Sources/ChatUI/ChatInChannel/MessageField/Giphy/View/GiphyAttributionMarkView.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+    
+    private func getImageName() -> String {
+        colorScheme == .dark
+        ? "Poweredby_100px-Black_VertText"
+        : "Poweredby_100px-White_VertText"
+    }
+}
+
```
Comment (x-0o0) [inline Sources/ChatUI/ChatInChannel/MessageField/Giphy/View/GiphyAttributionMarkView.swift:32]: Please add `GiphyAttributionMarkView.Previews` to `Previews/ChatInChannel` path:
```swift
struct GiphyAttributionMarkView_Preview: PreviewProvider {
    static var previews: some View {
        GiphyAttributionMarkView()
            .preferredColorScheme(.dark)
    }
}
```

**Comment by x-0o0** [comment]
Hi @fred-bowker :) Sorry for the late check.

What a good catch! 
Thanks for the great PR 👍

However, while reviewing the PR, I faced an *image unassigned issue* and the image doesn't show up on the preview canvas:
Screenshot 2023-10-01 at 8 10 11 PM

Also could you add the previews of the new view:`GiphyAttributionMarkView` to `Previews/ChatInChannel` path?

Screenshot 2023-10-01 at 8 20 32 PM
